gym-unity/gym_unity/envs/__init__.py

```diff
@@ -528,8 +531,12 @@ def mark_agent_done(self, agent_id: int, reward: float) -> None:
         """
         Declare the agent done with the corresponding final reward.
         """
-        gym_index = self._agent_id_to_gym_index.pop(agent_id)
-        self._done_agents_index_to_last_reward[gym_index] = reward
+        if agent_id in self._agent_id_to_gym_index:
+            gym_index = self._agent_id_to_gym_index.pop(agent_id)
```

Comment:
This is where the reported exceptions are. We could also make a check before the call to `mark_agent_done`, but it seemed a bit cleaner to do it here.